### DISCUSSION TITLE
Show a big warning, if `direnv allow` would be needed.

### DISCUSSION THREAD

**Discussion by guettli** [starship/starship#5230] [General]
I use [direnv](//direnv.net)

```
❯ cd -
/home/guettli/syself/foo
direnv: error /home/guettli/syself/foo/.envrc is blocked. Run `direnv allow` to approve its content
```

Sometimes I don't notice this warning, and then things don't work the way I expect them to - and I waste time.

Is there a Starship plugin which warns me if I need to call `direnv allow`?

**Comment by mickimnet** [comment]
Currently, there is no support for `direnv`. There is a pull request to add a `direnv` module, which includes the option for the state where you would need to run `direnv allow`—see: #5157

**Reply by guettli** [reply]
Thank you for the answer. This would be exactly what i was looking for.

**Reply by guettli** [reply]
I am trying to find a solution: https://www.reddit.com/r/commandline/comments/14wt75g/starship_prompt_show_warning_if_direnv_allow_is/

**Reply by guettli** [reply]
I found this solution (which works today, without an official direnv plugin):

# starship.toml
```
[custom.direnv]
command = '[[ $(direnv status) =~ "Found RC allowed false" ]] && echo "=========> missing: direnv allow <============"' # shows output of command
detect_files = ['.envrc'] # can specify filters but wildcards are not supported
```

**Reply by john-unger** [reply]
I was going to provide an update for this custom prompt. However after further research it turns out [starship already has a direnv config now](https://starship.rs/config/#direnv). So this is the config I have to only prompt me when this is needed, but hidden otherwise:

```yaml

[direnv]
disabled = false
symbol = ''  # No global symbol
allowed_msg = ''            # Hide "OK" state
loaded_msg = ''             # Hide "OK" state
not_allowed_msg = ' 🚫 not allowed'
denied_msg = ' 🔒 denied'
unloaded_msg = ' 🌪 not loaded'
format = '[$symbol$loaded$allowed]($style) '
style = 'bold yellow'
```

**Reply by guettli** [reply]
@john-unger thank you for the hint. I use your config now, too. There is one typo in your last line. The trailing ` char needs to be removed.
